packages/engine/src/datastore/storage/memory.rs

```diff
@@ -129,7 +129,7 @@ impl Memory {
         })
     }
 
-    pub fn from_message(
+    pub fn shmem_os_id(
```

Comment:
Not necessarily in the scope of this PR:
While in IDE this may be clear, but when reviewing the code, it's hard to tell, what the booleans mean, such as in 
```rust
Memory::shmem_os_id(os_id, true, true);
```
It's common in Rust to use `enum`s instead.